benchexec/containerized_tool.py

```diff
@@ -124,6 +124,14 @@ def _init_container_and_load_tool(tool_module, *args, **kwargs):
     try:
         _init_container(*args, **kwargs)
     except OSError as e:
+        if (
+            util.try_read_file("/proc/sys/kernel/apparmor_restrict_unprivileged_userns")
+            == "1"
+        ) and e.errno in [
+            errno.EPERM,
+            errno.EACCES,
+        ]:
+            logging.critical(container._ERROR_MSG_USER_NS_RESTRICTION)
```

Comment:
Here we could instead easily raise a `BenchExecException` with the error message, right? Then at least for users of `benchexec` we would get the nice way of showing the error message, only for `runexec` and `containerexec` not.

Reply:
That's true, I'll raise a `BenchExecException` here. The output would be this:

```
(benchexec-venv) benchexec-venvhaoranyang@haoranyang-fudan:~/benchexec$ benchexec doc/benchmark-example-rand.xml --tasks "XML files" --limitCores 1 --timelimit 10s --numOfThreads 4
2024-06-12 13:28:05 - WARNING - Creating UID mapping into container failed: [Errno 1] Operation not permitted
2024-06-12 13:28:05 - WARNING - Could not write to setgroups file in /proc: [Errno 13] Permission denied: '/proc/71549/setgroups'
2024-06-12 13:28:05 - WARNING - Creating GID mapping into container failed: [Errno 1] Operation not permitted
Error: Unprivileged user namespaces forbidden on this system, please enable them with 'sysctl -w kernel.apparmor_restrict_unprivileged_userns=0'. Ubuntu disable it by default since 24.04, refer to https://ubuntu.com/blog/whats-new-in-security-for-ubuntu-24-04-lts for more information.
```